interfaces.go

```diff
@@ -93,10 +93,13 @@ func (b *Breadcrumb) MarshalJSON() ([]byte, error) {
 // User describes the user associated with an Event. If this is used, at least
 // an ID or an IP address should be provided.
 type User struct {
-	Email     string `json:"email,omitempty"`
-	ID        string `json:"id,omitempty"`
-	IPAddress string `json:"ip_address,omitempty"`
-	Username  string `json:"username,omitempty"`
+	Data      map[string]string `json:"data,omitempty"`
```

Comment:
We should not add `Data` since it is not part of https://develop.sentry.dev/sdk/event-payloads/user/

the event payloads json schema contain some internal types, but we shouldn’t expose all of it. In this case I don’t think we need to expose `Data`

Reply:
In #466, they pointed to https://develop.sentry.dev/sdk/event-payloads/types/#typedef-User, which contains the `user.data`. So this should be considered `internal`?

@marandaneto Abhi told me you did some research about this.

Reply:
`other` is internal but `data` should be public, Mobile SDKs have the `data` as well.